### PR TITLE
Partially enable trailing commas lint. Fixed about 200 out of 600 violations

### PR DESCRIPTION
**Pull request by jacob314**
RELEASE_NOTE_EXCEPTION=no changes in functionally. Just lots of commas.
Fixed all violations in non test code and some of the test violations.
In general, some of the trailing commas don't impact formatting (arrays already use flutter style without a trailing comma) and the violations from test code seem less controversial as they seem aligned with @kenzieschmoll code review comments).

### CODE REVIEW

File: packages/devtools_app/lib/src/framework/app_error_handling.dart [hunk 1]
```diff
@@ -22,24 +22,26 @@ import '../shared/config_specific/logger/logger.dart';
...
-
-      if (oldHandler != null) {
-        oldHandler(details);
-      }
-    };
-
-    return appStartCallback();
-  }, (Object error, StackTrace stack) {
-    _reportError(error, stack);
-  });
```
Comment (jacob314) [inline packages/devtools_app/lib/src/framework/app_error_handling.dart:42]: Example of fixes that I think match what @kenzieschmoll would suggest in a code review.

File: packages/devtools_app/lib/src/framework/status_line.dart [hunk 2]
```diff
@@ -186,7 +186,7 @@ class StatusLine extends StatelessWidget {
                           style: textTheme.bodyMedium,
                           overflow: TextOverflow.clip,
                         ),
-                      ]
```
Comment (jacob314) [inline packages/devtools_app/lib/src/framework/status_line.dart:189]: Example where a trailing comma doesn't change what dartfmt does but where it does seem fairly consistent with how we are using trailing commas.

File: packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart [hunk 3]
```diff
@@ -743,7 +743,7 @@ extension AppSizeJsonFileExtension on DevToolsJsonFile {
     'macos',
     'windows',
     'linux',
-    'web'
+    'web',
```
Comment (jacob314) [inline packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart:746]: this case isn't consistent with how we use trailing commas but seems harmless enough. The best argument in favor for this case is just that the trailing commas make the diffs smaller if you happen to edit the List later.

**Review by jacob314 — COMMENTED** [review]
Fyi @pq for whether this gives you insights on how we might want to tweak the trailing comma support in the Dart linter. We wouldn't want to require trailing commas at the end of all List objects but some of the other cases do help reduce extra nesting by dartfmt and seem generally aligned with Flutter style.

File: packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart [hunk 4]
```diff
@@ -44,12 +46,12 @@ class _TraceCheckBoxColumn extends ColumnData<TracedClass>
...
     return Checkbox(
       value: item.traceAllocations,
-      onChanged: (value) async {
+      onChanged: (value) {
         ga.select(
           gac.memory,
           '${gac.MemoryEvent.tracingTraceCheck}-$value',
         );
-        await controller.setAllocationTracingForClass(item.cls, value!);
+        unawaited(controller.setAllocationTracingForClass(item.cls, value!));
```
Comment (CoderDake) [inline packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart:54]: is this comma related?

**Review by CoderDake — APPROVED** [review]
LGTM.
You'll need to add a release note exception mentioning that this isn't user facing :)

**Comment by pq** [comment]
> Fyi @pq for whether this gives you insights on how we might want to tweak the trailing comma support in the Dart linter. We wouldn't want to require trailing commas at the end of all List objects but some of the other cases do help reduce extra nesting by dartfmt and seem generally aligned with Flutter style.

Interesting.  Thanks!  It looks like we can probably settle on a pretty simple heuristic and it could really cut down on code review churn.

Hilariously, such a lint would have avoided this exchange just the other day!

https://dart-review.googlesource.com/c/sdk/+/276770/12..17/pkg/analysis_server/lib/src/services/correction/fix_internal.dart#b1444

File: packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart (hunk 4 above)
Comment (jacob314) [reply packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart]: oops. reverted. I really need to turn on auto-save in VSCode. That was supposed to be for a separate CL but I must have left the file unsaved and then hit save-all.

**Comment by jacob314** [comment]
> https://dart-review.googlesource.com/c/sdk/+/276770/12..17/pkg/analysis_server/lib/src/services/correction/fix_internal.dart#b1444

Ah interesting. @bwilkerson for the analyzer code base would you be ok with requiring trailing commas at the end of more list literals matching what this lint is doing?

**Comment by bwilkerson** [comment]
My comment wasn't related to wanting to use trailing commas everywhere, it was about wanting to be consistent within a single map/list literal. I don't think I'd find a lot of value to enabling trailing_commas, but I also wouldn't object too much if other team members really want it. My biggest objection (aside from a strong belief that the formatter should just do a better job of formatting without requiring trailing commas) is that we don't have a corresponding check to tell us when we should remove trailing commas.
